compair/models/assignment.py

```diff
@@ -168,6 +172,50 @@ def after_comparing(self):
         else:
             return now >= self.compare_end.replace(tzinfo=pytz.utc)
 
+    @hybrid_property
+    def self_eval_period(self):
+        now = dateutil.parser.parse(datetime.datetime.utcnow().replace(tzinfo=pytz.utc).isoformat())
```

Comment:
Think this is slightly off
```
now = dateutil.parser.parse(datetime.datetime.utcnow().replace(tzinfo=pytz.utc).isoformat())
if not self.enable_self_evaluation:
    return False
elif not self.self_eval_start:
    compare_start = self.compare_start.replace(tzinfo=pytz.utc)
    answer_end = self.answer_end.replace(tzinfo=pytz.utc)
    if self.compare_start:
        return compare_start <= now
    else:
        return answer_end <= now
else:
    return self.self_eval_start.replace(tzinfo=pytz.utc) <= now < self.self_eval_end.replace(tzinfo=pytz.utc)
```

Reply:
Thanks, good idea to add in `enable_self_evaluation`. I think this better covers it, though: 
```python
        if not self.enable_self_evaluation:
            return False
        elif not self.self_eval_start:
            return self.compare_period
        else:
            return self.self_eval_start.replace(tzinfo=pytz.utc) <= now < self.self_eval_end.replace(tzinfo=pytz.utc)
```

Reply:
If you use compare period, then self evaluations will close when comparisons close. Not sure if this is desired. @lenglund thoughts?

Reply:
In general, the request from the instructor side is for self-evals to exist independent of all other work being done (the thought being students submit and compare and read feedback first, then have time to reflect / self-eval later). Does that answer the question?

Reply:
This is the current functionality. If the instructor wants to give students that extra flexibility, they can explicitly set the self-eval dates, in which case the self-eval is not affected by comparison end date. If the self-eval dates are not set, however, I think it makes sense to keep the current logic. (I thought that's what we had decided already?)

Reply:
I guess my only hold up is that if there is no self-eval date set, then it shouldn't ever end (whether there is a comparison period or not). That's how comparisons handle having no dates set

Reply:
But that is not the logic currently. Currently, if there is no self-eval date set, self-eval ends at the same time as the comparison end. Instructors still will have the option to set the self-eval date to go until the end of the course by setting it explicitly. I am thinking maybe something we can do is default the self-eval end date to the course end date (currently it's not set), so all they would have to do is check the box and it will set it for them. 

At the end of the day, I think this just comes down to a decision. Do we want to change the existing logic even when they are not setting self-eval dates?